src/Misc/MiddleWare.cpp

```diff
@@ -271,30 +271,31 @@ static void idsFromMsg(const char* msg, int* part, int* kit, int* voice)
  */
 class WaveTableRequestHandler
 {
-    bool waitForAdPars[NUM_MIDI_PARTS][NUM_KIT_ITEMS][NUM_VOICES];
+    bool waitForAdPars[NUM_MIDI_PARTS][NUM_KIT_ITEMS][NUM_VOICES][2];
 
 public:
     WaveTableRequestHandler()
     {
         for(std::size_t p = 0; p < NUM_MIDI_PARTS; ++p)
         for(std::size_t k = 0; k < NUM_KIT_ITEMS; ++k)
         for(std::size_t v = 0; v < NUM_VOICES; ++v)
-            waitForAdPars[p][k][v] = false;
+        for(std::size_t i = 0; i < 2; ++i)
+            waitForAdPars[p][k][v][i] = false;
     }
 
-    void chainWtParamRequest(int part, int kit, int voice, rtosc::RtData& d)
+    void chainWtParamRequest(int part, int kit, int voice, bool isFm, rtosc::RtData& d)
```

Comment:
While a minor convention, it might make sense to refer to the secondary oscillator as a modulation source rather than the FM oscillator. I think originally only FM was in zyn which lead to the internal naming convention, but at this point it's just a fairly arbitrary modulation source.

Reply:
Thanks for the reminder. I'll fix this at the end of this PR and leave your comment open to be reminded.